radicle/src/ssh.rs

```diff
@@ -75,30 +77,78 @@ pub mod fmt {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum SignatureError {
+    #[error(transparent)]
+    Invalid(#[from] crypto::Error),
+    #[error(transparent)]
+    Encoding(#[from] encoding::Error),
+    #[error("unknown algorithm '{0}'")]
+    UnknownAlgorithm(String),
+}
+
+impl Encodable for crypto::Signature {
+    type Error = SignatureError;
+
+    fn read_ssh(r: &mut encoding::Cursor) -> Result<Self, Self::Error> {
+        let buf = r.read_string()?;
+        let mut r2 = buf.reader(0);
```

Comment:
I think @cloudhead mentioned something about the numbered vars. You could just do:
```suggestion
        let mut buf = buf.reader(0);
```

Since `buf` isn't really used again anyway :shrug:

Reply:
Cool, but instead used the name 'inner_strs' to use consistent naming w/ read(..)